sources/packages/backend/apps/api/src/route-controllers/cas-invoice/tests/e2e/cas-invoice.aest.controller.getInvoices.e2e-spec.ts

```diff
@@ -0,0 +1,271 @@
+  BEARER_AUTH_TYPE,
+  createTestingAppModule,
+  getAESTToken,
+} from "../../../../testHelpers";
+import {
+  createE2EDataSources,
+  createFakeCASInvoiceBatch,
+  createFakeDisbursementValue,
+  E2EDataSources,
+  saveFakeInvoiceFromDisbursementReceipt,
+  saveFakeInvoiceIntoBatchWithInvoiceDetails,
+} from "@sims/test-utils";
+import { SystemUsersService } from "@sims/services";
+import {
+  CASInvoiceStatus,
+  DisbursementValueType,
+  OfferingIntensity,
+  SupplierStatus,
+} from "@sims/sims-db";
+import { CASInvoiceAPIOutDTO } from "apps/api/src/route-controllers/cas-invoice/models/cas-invoice.dto";
+import { addDays } from "@sims/utilities";
+import { Between } from "typeorm";
+
+/**
+ * Use a period that will never be reached to delete all existing invoice batches
+ * and allow the retrieval of invoice batches to be tested.
+ */
+const CAS_INVOICE_STATUS_LAST_UPDATED_ON_START_DATE = new Date("2080-01-01");
+const CAS_INVOICE_STATUS_LAST_UPDATED_ON_END_DATE = new Date("2080-02-01");
+
+describe("CASInvoiceAESTController(e2e)-getInvoices", () => {
+  let app: INestApplication;
+  let db: E2EDataSources;
+  let systemUsersService: SystemUsersService;
+
+  beforeAll(async () => {
+    const { nestApplication, dataSource } = await createTestingAppModule();
+    app = nestApplication;
+    systemUsersService = app.get(SystemUsersService);
+    db = createE2EDataSources(dataSource);
+  });
+
+  beforeEach(async () => {
+    // Delete existing CAS invoices and invoice details created between the CAS_INVOICE_STATUS_LAST_UPDATED_ON_START_DATE and the CAS_INVOICE_STATUS_LAST_UPDATED_ON_END_DATE dates.
+    const invoicesToDelete = await db.casInvoice.find({
+      select: { id: true },
+      where: {
+        invoiceStatusUpdatedOn: Between(
+          CAS_INVOICE_STATUS_LAST_UPDATED_ON_START_DATE,
+          CAS_INVOICE_STATUS_LAST_UPDATED_ON_END_DATE,
+        ),
+      },
+    });
+    for (const invoice of invoicesToDelete) {
+      await db.casInvoiceDetail.delete({
+        casInvoice: invoice,
+      });
+    }
+    await db.casInvoice.delete({
+      invoiceStatusUpdatedOn: Between(
+        CAS_INVOICE_STATUS_LAST_UPDATED_ON_START_DATE,
+        CAS_INVOICE_STATUS_LAST_UPDATED_ON_END_DATE,
+      ),
+    });
+  });
+
+  it(`Should be able to get invoices for the first page in a paginated result with a limit of two per page with ${CASInvoiceStatus.ManualIntervention} status in the descending order.`, async () => {
+    // Arrange
+    // Create CAS invoice batch.
+    const casInvoiceBatch = await db.casInvoiceBatch.save(
+      createFakeCASInvoiceBatch({
+        creator: systemUsersService.systemUser,
+      }),
+    );
+    // Creates part-time application with receipts, and invoices details.
+    const casInvoice = await saveFakeInvoiceIntoBatchWithInvoiceDetails(
+      db,
+      {
+        casInvoiceBatch,
+        creator: systemUsersService.systemUser,
+        // Part-time BC grants.
+        disbursementValues: [
```

Comment:
A note for the future: unless required for the test, we can avoid creating more records than required. Here, one `DisbursementValueType.BCGrant` would be enough.